test/engine-luatest/gh_6436_complex_foreign_key_test.lua

```diff
@@ -63,6 +63,11 @@ g.test_bad_complex_foreign_key = function(cg)
             "Illegal parameters, foreign key: foreign field must be string or number",
             function() box.schema.create_space('city', opts) end
         )
+        opts = space_opts({['']={space='country',field={p_id='planet_id', c_id='country_id'}}})
+        t.assert_error_msg_content_equals(
+            "Invalid identifier '' (expected printable symbols only or it is too long)",
```

Comment:
I believe you do not check name length:
```
tarantool> name = string.rep('a', 1000000)
---
...

tarantool> #name
---
- 1000000
...

tarantool> s = box.schema.space.create('t', {format = {{'a', 'integer'}, {'b', 'integer', foreign_key = {[name] = {field = 'a'}}}}})
---
...

tarantool> for k, v in pairs(s:format()[2].foreign_key) do print(#k) end
1000000
---
...
```

Reply:
I think it's better to place these checks here, together with other variants of incorrect constraints.

Name length check is added.